unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/ECSRuntime/Components/ECSComponent.cs

```diff
@@ -0,0 +1,144 @@
+using System;
+using System.Collections.Generic;
+using DCL.Controllers;
+using DCL.Models;
+using UnityEngine;
+
+namespace DCL.ECSRuntime
+{
+    public class ECSComponent<ModelType> : IECSComponent
+    {
+        internal Dictionary<string, ECSComponentData<ModelType>> entities = new Dictionary<string, ECSComponentData<ModelType>>();
```

Comment:
I agree with that. For me, it is more natural to have entities that reference components rather than the other way.

If that is temporal it is ok for me!

Reply:
no, this PR is not "for the time being". With that I meant no having an enum for components ids, not implementing usage for the factory added here, etc
yes, we are using entities as "weak" actor, we are treating almost as just indexes in a dictionary (maybe an array in the future) of data so we can have a decoupled with the minimum amount of dependencies system as we can.
having components references inside an entity leaves `entity -> component -> entity` and here we are going for `component -> entity`.
also it can be aimed for a better performance... updating all `Transform` components together could increase cache hits, while moving from entity to component has to many jumps in memory, so it give as the possibility for those kind of improvements and flexibility since we don't have any dependencies on entities

Reply:
I've read a bunch of documentation on the matter (to be honest I'm not an ECS expert) and I'm not seeing components referencing entities in any of the implementations of ECS I've come across. 

The closer would be the approach explained here: https://www.david-colson.com/2020/02/09/making-a-simple-ecs.html.
It uses bitmasks to map the relationships between entities and components but even then, components are just plain data structures. They do not contain much logic and no reference to the entities, that relationship is mapped in the scene itself.

The DOTS approach also follows a similar pattern but using archtypes to map entities with the same components to faster iterate on them: https://docs.unity3d.com/Packages/com.unity.entities@0.50/manual/ecs_core.html.

To be honest I don't have an strong opinion about this, it kinda makes sense to have components referencing entities because that's usually the use case ("update me and all the entities I'm attached to") but I'm afraid we might be missing something when every implementation I've found is doing it the other way around.

Reply:
We agreed to leave it as it is for the time being. If we see components starting to increase in complexity we will gather them in a system to stablish the relation with entities.